File: runtime/Swift/Sources/Antlr4/atn/ActionTransition.swift
/// 
/// Copyright (c) 2012-2017 The ANTLR Project. All rights reserved.
/// Use of this file is governed by the BSD 3-clause license that
/// can be found in the LICENSE.txt file in the project root.
/// 


public final class ActionTransition: Transition, CustomStringConvertible {
    public let ruleIndex: Int
    public let actionIndex: Int
    public let isCtxDependent: Bool
    // e.g., $i ref in action


    public convenience init(_ target: ATNState, _ ruleIndex: Int) {
        self.init(target, ruleIndex, -1, false)
    }

    public init(_ target: ATNState, _ ruleIndex: Int, _ actionIndex: Int, _ isCtxDependent: Bool) {

        self.ruleIndex = ruleIndex
        self.actionIndex = actionIndex
        self.isCtxDependent = isCtxDependent
        super.init(target)
    }

    override
    public func getSerializationType() -> Int {
        return Transition.ACTION
    }

    override
    public func isEpsilon() -> Bool {
        return true // we are to be ignored by analysis 'cept for predicates
    }

    override
    public func matches(_ symbol: Int, _ minVocabSymbol: Int, _ maxVocabSymbol: Int) -> Bool {
        return false
    }

    public var description: String {
        return "action_\(ruleIndex):\(actionIndex)"
    }

}

File: runtime/JavaScript/src/antlr4/transition/ActionTransition.js
/* Copyright (c) 2012-2022 The ANTLR Project Contributors. All rights reserved.
 * Use is of this file is governed by the BSD 3-clause license that
 * can be found in the LICENSE.txt file in the project root.
 */
import Transition from "./Transition.js";

export default class ActionTransition extends Transition {
    constructor(target, ruleIndex, actionIndex, isCtxDependent) {
        super(target);
        this.serializationType = Transition.ACTION;
        this.ruleIndex = ruleIndex;
        this.actionIndex = actionIndex===undefined ? -1 : actionIndex;
        this.isCtxDependent = isCtxDependent===undefined ? false : isCtxDependent; // e.g., $i ref in pred
        this.isEpsilon = true;
    }

    matches(symbol, minVocabSymbol, maxVocabSymbol) {
        return false;
    }

    toString() {
        return "action_" + this.ruleIndex + ":" + this.actionIndex;
    }
}
